### PR TITLE
Fix checkbox alignment issues in options and block manager modals (#16860)

### LINKED ISSUE #16860: Misplaced checkmarks in the options modal
I'm seeing some misplaced checkmarks in the Options modal. Tested using `master` in the latest versions of Safari, FF, and Chrome on MacOS: 

Screen Shot 2019-08-01 at 10 01 31 AM

Likely related to #16551.

cc @youknowriad + @jffng

### PR DESCRIPTION
**Pull request by jffng**
## Description
Checkboxes are misaligned in the options + block manager modals due to an attempt to replace them with SVG (#16551).

This PR introduces editor specific css to address those alignment issues. 

## How has this been tested?
Tested MacOS on latest Chrome, Firefox, and Safari. 

## Screenshots 
Screen Shot 2019-08-01 at 12 23 25 PM
Screen Shot 2019-08-01 at 12 23 15 PM

## Types of changes
Fixes #16860

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 

cc @kjellr

### CODE REVIEW

**Comment by kjellr** [comment]
Thanks for the quick fix, @jffng. Just a small nitpicky note: 

I'm seeing a little bit of spacing inconsistency, and I can't figure out where it's coming from. Check out these two adjacent list items in the Options panel for example: 

Screen Shot 2019-08-01 at 1 58 18 PM

Notice that the first checkmark is appearing higher than the one below it. Those should theoretically have the exact same styles, so I'm not sure what's causing the inconsistency. Any ideas?

File: packages/components/src/checkbox-control/style.scss
Comment (youknowriad) [inline packages/components/src/checkbox-control/style.scss]: Do you think it's possible to replace this media query without our media query mixins?

**Comment by jffng** [comment]
@kjellr it looks like the actual inputs are being rendered in ever so slightly different positions, thus throwing off the position of the checkmark relative to the box: 

Screen Shot 2019-08-02 at 10 55 15 AM

The issue goes away when the width + height of the input is greater than some threshold, e.g. 19 x 19px:

Screen Shot 2019-08-02 at 10 58 41 AM

File: packages/edit-post/src/components/manage-blocks-modal/style.scss
Comment (kjellr) [inline packages/edit-post/src/components/manage-blocks-modal/style.scss]: Really minor comment: but we should use our [`_z-index.scss` mixin](https://github.com/WordPress/gutenberg/blob/master/assets/stylesheets/_z-index.scss) for this. Just add a ...  

`".edit-post-manage-blocks-modal__category-title": 1,` 

... entry in that file, and replace this line with: 

`z-index: z-index(".edit-post-manage-blocks-modal__category-title");`

**Comment by kjellr** [comment]
> it looks like the actual inputs are being rendered in ever so slightly different positions, thus throwing off the position of the checkmark relative to the box:

That is truly bizarre. 🤔 In any case, I don't think it should hold up this PR. I posted a tiny comment about the `z-index` above. Once that's taken care of this should be good to ship!

**Comment by jffng** [comment]
@kjellr I think I came up with a fix. The height of the labels was fractionally different in some places due to a `rem` padding calculation:

Screen Shot 2019-08-02 at 4 06 14 PM

Manually setting the height of the label seems to have done the trick. 

Mind giving it one more spin to see if passes the eye test? 

Screen Shot 2019-08-02 at 4 17 36 PM

**Comment by youknowriad** [comment]
It's a bit concerning that the height of the label affects the checkbox, I was expecting that the checkbox component would work properly no matter the context it's being used in? Does this mean any developer using the component would have to adjust the height of the labels somehow?

**Comment by jffng** [comment]
@youknowriad this is a valid concern. I pushed an update that should address it, let me know if it's okay. Previously I wanted to avoid adding any additional markup to the component, but looking at other checkbox implementations, I think it's reasonable to add a span that wraps the input and svg elements in a single inline container.

**Comment by kjellr** [comment]
I'll let Riad weigh in on the method, but I can confirm that this fixes the spacing issue. Visually, this looks good! 

Screen Shot 2019-08-05 at 11 39 23 AM

Screen Shot 2019-08-05 at 11 39 10 AM

File: packages/components/src/checkbox-control/style.scss
Comment (kjellr) [inline packages/components/src/checkbox-control/style.scss]: Should these be declared somewhere more global (ie. [`_variables.scss`](https://github.com/WordPress/gutenberg/blob/master/assets/stylesheets/_variables.scss))?

File: packages/edit-post/src/components/manage-blocks-modal/style.scss
Comment (youknowriad) [inline packages/edit-post/src/components/manage-blocks-modal/style.scss]: How is this change related?

**Comment by youknowriad** [comment]
The approach seems good to me. I'm wondering why do we need changes outside the component itself?

**Comment by youknowriad** [comment]
(Sorry, clicked the wrong button)(Sorry, clicked the wrong button)

File: packages/edit-post/src/components/manage-blocks-modal/style.scss
Comment (jffng) [reply packages/edit-post/src/components/manage-blocks-modal/style.scss]: The headings for the block manager modal use `position: sticky`, creating a "sticky container" parent. Using a `position: relative` element (the wrapper for the input to position the checkbox) within the same sticky parent container, the browser treats them as in the same stacking context: 

Screen Recording 2019-08-05 at 03 17 PM

Adding a z-index to the header makes the sticky header remain on top:

Screen Recording 2019-08-05 at 03 20 PM

File: assets/stylesheets/_variables.scss
Comment (youknowriad) [inline assets/stylesheets/_variables.scss]: To be honest, I'm not certain about these variables being global here. I think it only make sense if we use them consistently in all "inputs" otherwise it's going to create confusion. For example, why the size of the TextControl component doesn't use it?...

Thoughts @kjellr maybe we can leave them in the component for now, and maybe rename them to make it clear that it's just about the checkbox?

Comment (kjellr) [reply assets/stylesheets/_variables.scss]: Yeah, I just conferred with @jasmussen about this, and while we haven't usually declared variables within components in the past, we can see how it would make sense to allow them.

Feel free to move these back, @jffng (and sorry for the back and forth!). Might also want to make the names just a little more specific too, as Riad noted. 

Thanks!

**Comment by youknowriad** [comment]
Looks like there are some failing unit tests preventing us from merging this branch. Is this something you can take care of or need help with?

**Comment by jffng** [comment]
Thanks @youknowriad, this link helped me out for updating the snapshots: 

https://developer.wordpress.org/block-editor/contributors/develop/testing-overview/#tldr-broken-snapshots

**Review by youknowriad — APPROVED** [review]
Thanks for the updates. LGTM 👍

**Comment by kjellr** [comment]
🎉 Thanks for sorting this out, @jffng!

**Pull request merged**
